lib/Sema/PreCheckExpr.cpp

```diff
@@ -1056,6 +1056,29 @@ namespace {
           if (isa<SequenceExpr>(parent))
             return finish(true, expr);
 
+          // Unwrap to the outermost paren in the sequence.
+          if (isa<ParenExpr>(parent)) {
+            for (;;) {
+              auto nextParent = parents.find(parent);
+              if (nextParent == parents.end())
+                break;
+
+              // e.g. `foo((&bar), x: ...)`
+              if (isa<TupleExpr>(nextParent->second)) {
+                parent = nextParent->second;
+                break;
+              }
+
+              // e.g. `foo(((&bar))`
+              if (isa<ParenExpr>(nextParent->second)) {
```

Comment:
Should this try to look through all `IdentityExpr`s?

Reply:
I think in this case we really want to only look through parens, until `!` where we could ignore a `try` or `await`.